contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/afterSimAnalysis/StoppingTaskRecorder.java

```diff
@@ -48,80 +62,73 @@ public double getEndTime() {
 			return endTime;
 		}
 
-		public Id<Person> getPersonId() {
-			return personId;
+		public Id<DvrpVehicle> getVehicleId() {
+			return vehicleId;
 		}
 
-		public String getTaskId() {
-			return taskId;
+		public int getOccupancy() {
+			return occupancy;
 		}
-
-		public Id<Link> getLinkId() {
-			return linkId;
-		}
-
 	}
 
 	@Override
-	public void handleEvent(ActivityStartEvent event) {
-		if (event.getActType().equals("DrtStay")) {
-			Id<Person> personId = event.getPersonId();
-			DrtStoppingTaskDataEntry stayTaskDataEntry = new DrtStoppingTaskDataEntry("stayTask_" + stayTaskCounter,
-					personId, event.getTime(), event.getLinkId());
-			startedStayTasksMap.put(personId, stayTaskDataEntry);
-			stayTaskCounter += 1;
+	public void handleEvent(TaskStartedEvent taskStartedEvent) {
+		Id<DvrpVehicle> vehicleId = taskStartedEvent.getDvrpVehicleId();
+		Id<Link> linkId = taskStartedEvent.getLinkId();
+		double time = taskStartedEvent.getTime();
+
+		// Stay task
+		if (taskStartedEvent.getTaskType().equals(DrtStayTask.TYPE)) {
```

Comment:
do you mean any drt stay task (`DrtTaskBaseType.STAY`) or exactly `DrtStayTask.TYPE` ?

Reply:
I think I just need the STAY task

Reply:
The exact stay task. It does not include Stop task.

Reply:
But then you should check: `DrtTaskBaseType.STAY.isBaseTypeOf(taskStartedEvent.getTaskType))` instead of `taskStartedEvent.getTaskType().equals(DrtStayTask.TYPE)`